atomix/cluster/src/test/java/io/atomix/raft/RaftStartupConsistencyCheckTest.java

```diff
@@ -73,4 +75,36 @@ public void shouldNotFailRestartIfFollowerCrashedBeforeCommittingSnapshot() thro
                 assertThat(raftRule.getPersistedSnapshotStore(followerId).getCurrentSnapshotIndex())
                     .isEqualTo(snapshotIndex));
   }
+
+  @Test // regression test for https://github.com/camunda/zeebe/issues/14367
+  public void shouldHandleRetriedRequestsAfterSnapshotPersist() throws Exception {
+    // given -- force a follower to receive a snapshot on restart
+    final var leader = raftRule.getLeader().orElseThrow();
+    final var follower = raftRule.getFollower().orElseThrow();
+    raftRule.shutdownServer(follower);
+
+    final var snapshotIndex = raftRule.appendEntries(500);
+    raftRule.takeCompactingSnapshot(leader, snapshotIndex - 1, 3);
+    final var lastIndex = raftRule.appendEntries(10);
+    leader.getContext().setPreferSnapshotReplicationThreshold(1);
+
+    // when -- persisting the snapshot on the follower is too slow, the leader will retry the
+    // `install` requests
+    final Runnable interceptor =
+        () -> {
+          try {
+            // TODO: better way to force a timeout?
```

Comment:
😆 are you planning to do something here?

Reply:
I was hoping you'd have a better idea :smile:

Reply:
I added a way to intercept requests in the test protocol so we don't need to sleep.